pkg/api/vlabs/validate.go

```diff
@@ -998,44 +994,6 @@ func (a *Properties) validateServicePrincipalProfile() error {
 	return nil
 }
 
-func (a *Properties) validateManagedIdentity() error {
```

Comment:
This entire validation logic is to ensure you're using k8s >= 1.12.0, which is not necessary.